packages/itwinui-css/src/input-container/input-container.scss

```diff
@@ -106,39 +98,34 @@
 // ----------------------------------------------------------------------------
 
 .iui-input-flex-container {
-  --_iui-input-background-color: var(--iui-color-background);
-  --_iui-input-border-color: var(--iui-color-border-foreground);
   --_iui-input-accent-border-color: var(--iui-color-border-accent);
-  --_iui-input-text-color: var(--iui-color-text);
   --_iui-input-padding-block: var(--iui-size-2xs);
-  --_iui-input-min-height: var(--iui-component-height);
 
   display: inline-flex;
   align-items: center;
-  background-color: var(--_iui-input-background-color);
-  border-radius: var(--iui-border-radius-1);
-  color: var(--_iui-input-text-color);
   position: relative;
-  block-size: var(--_iui-input-min-height);
   inline-size: 100%;
   isolation: isolate;
-  min-block-size: var(--_iui-input-min-height);
   block-size: fit-content;
   line-height: var(--iui-size-l);
+  border: none;
```

Comment:
This looks a bit random. What's the intent here? I'm honestly pretty confused looking at this file, and there is no useful explanation in the PR description/prior comments.

Reply:
Ah, good question.

`.iui-field` defines the CSS variables and sets those variables on their respective properties such as `border`.
I added `.iui-field` to `.iui-input-container` so that we could use those CSS variables on the nested `<input>` and `<textarea>` (note that they do NOT use `.iui-input`).
However, the border is applied to those nested elements, NOT the `.iui-input-container` wrapper.  So this line removes the border from the wrapper.

Without this change, we have a double border:
![Screenshot 2024-04-24 at 1 19 13 PM](https://github.com/iTwin/iTwinUI/assets/849817/a2b6722e-e414-4354-8fa0-ae5fdc8d2fec)

Reply:
Reverted `.iui-field` being required for `.iui-input-container`.

Reply:
I'm still seeing some `iui-field` references in there. https://github.com/iTwin/iTwinUI/blob/55a8896a10abf201f1eae790b8552b845aa168a6/packages/itwinui-css/src/input-container/input-container.scss#L39-L40

And there are some visual differences too; did you verify those?

Reply:
Those lines appear to be within `.iui-input-grid` which nests `.iui-field .iui-input` within it.
`.iui-input-container` however does not use `.iui-field .iui-input` and instead targets the elements `<input>` & `<textarea>` directly.
So those lines I believe are expected to stay.

Reply:
The visual changes are bug fixes.  `.iui-input-container` was not setting the correct text colors but now is.  Credit to @rohan-kadkol for the screenshot.
![image](https://github.com/iTwin/iTwinUI/assets/849817/c23b0d7c-77fa-45f3-b439-92ea8e6d02aa)